### PR TITLE
Updated to new API methods (zips/create) 

### PR DESCRIPTION
**Pull request by fhc95**
And added unzipped downloading for single files.

### CODE REVIEW

File: src/put [hunk 1]
```diff
@@ -273,24 +276,31 @@ def save(parser, context, args):
...
-		time.sleep(1)
-	filename = str(os.getppid()) + ".zip"
-	download_command = config_json["downloader"].replace("$1", cache_folder).replace("$2", filename).replace("$3", response_json["url"])
-	os.system(download_command)
-	os.system("unzip " + cache_folder + "/" + filename)
-	os.system("rm -f " + cache_folder + "/" + filename)
+
+	if len(file_ids) == 1 and api_call("files/" + file_ids[0])["file"]["file_type"] != "FOLDER":
+		download_url = api_call("files/" + file_ids[0] + "/download", None, True)
+		download_command = config_json["downloader"].replace("$1", ".").replace("$2", file_names[0]).replace("$3", download_url)
```
Comment (AgamAgarwal) [inline src/put]: Now that constructing the download_command is done in multiple places (here and line 299), I would recommend pulling it out into a function.

File: src/put [hunk 2]
```diff
@@ -273,24 +276,31 @@ def save(parser, context, args):
...
+		zip_id = response_json["zip_id"]
+		while 1:
+			response_json = api_call("zips/"+str(zip_id))
+			if "url" in response_json:
+				if response_json["url"] != None:
+					break
+			time.sleep(1)
+		filename = str(os.getppid()) + ".zip"
+		download_command = config_json["downloader"].replace("$1", cache_folder).replace("$2", filename).replace("$3", response_json["url"])
+		print(download_command)
```
Comment (AgamAgarwal) [inline src/put]: I am assuming this was for debugging. If yes, please remove this.

**Review by AgamAgarwal — CHANGES_REQUESTED** [review]
Hi Floris,
We are happy to see that someone other than us is actually using this tool. 😛 

Thanks a lot for the PR. Left a few minor comments. I will merge this PR once you resolve them.

Thanks a lot.

**Comment by fhc95** [comment]
Hi Agam,
Thanks for the feedback, I have updated my code.

I was looking at a few cli put.io projects on GitHub and this one seemed most up to date, I might add some other features in the future like searching.

**Review by AgamAgarwal — APPROVED** [review]

**Comment by AgamAgarwal** [comment]
Sure thing. Feel free to send more PRs.

**Pull request merged**